### PR TITLE
build(client): Update deps on build-tools to 0.39

### PR DESCRIPTION
**Pull request by tylerbutler**
Updated the following:

  client (release group)
  client-release-group-root

Dependencies on build-tools updated:

  @fluid-tools/build-cli: 0.39.0
  @fluidframework/build-tools: 0.39.0
  @fluidframework/bundle-size-tools: 0.39.0
  @fluid-tools/version-tools: 0.39.0

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -16903,8 +16903,8 @@ packages:
     dependencies:
       '@aws-crypto/sha256-browser': 3.0.0
       '@aws-crypto/sha256-js': 3.0.0
-      '@aws-sdk/client-sso-oidc': 3.569.0(@aws-sdk/client-sts@3.569.0)
-      '@aws-sdk/client-sts': 3.569.0
+      '@aws-sdk/client-sso-oidc': 3.569.0
+      '@aws-sdk/client-sts': 3.569.0(@aws-sdk/client-sso-oidc@3.569.0)
```
Comment (jason-ha) [inline pnpm-lock.yaml:16907]: These seem unfortunate (unstable?).

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -45689,7 +45672,7 @@ packages:
     resolution: {integrity: sha512-Y4X9zqrCftUhMeH2EptSSERdVKt/nEdijTOacGD/97EKjhQ/Qs8RTlEGABSJNNN8lac9kheH+af7yAkEWlgneA==}
     engines: {node: '>=10.0.0'}
     dependencies:
-      ajv: 8.12.0
+      ajv: 8.13.0
```
Comment (jason-ha) [inline pnpm-lock.yaml:45675]: collateral? probably okay but makes ajv (8.13.0 version) no longer a dev dep

**Comment by jason-ha** [comment]
CI complains with:
```
@fluidframework/core-interfaces: error during command 'flub generate typetests --dir . -v --publicFallback' (exit code 2)
@fluidframework/core-interfaces: ›   Error: Nonexistent flag: -v
@fluidframework/core-interfaces:  ›   See more help with --help
@fluidframework/driver-definitions: error during command 'flub generate typetests --dir . -v --publicFallback' (exit code 2)
@fluidframework/driver-definitions: ›   Error: Nonexistent flag: -v
@fluidframework/driver-definitions:  ›   See more help with --help
```
cc: @CraigMacomber

**Review by jason-ha — CHANGES_REQUESTED** [review]
I couldn't push to branch to apply fixes. So, I created derived branch and #21360

**Comment by tylerbutler** [comment]
Superseded by #21360; closing.

**Pull request closed without merging**